pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -414,10 +417,22 @@ func (configgen *ConfigGeneratorImpl) buildGatewayHTTPRouteConfig(node *model.Pr
 						vHost.RequireTls = route.VirtualHost_ALL
 					}
 				} else {
+					perRouteFilters := map[string]*anypb.Any{}
+					if svc, exists := nameToServiceMap[hostname]; exists {
```

Comment:
A bit confused how we are mapping hostname to service in the context of a gateway. The hostname will be like example.com typically?

Reply:
For gateways things are worse - we can't use a dedicated port ( it's very normal to have sessions on 80 and 443 ), so probably we'll need to have some annotation to enable sessions for the gateway - and something on the HttpRoute.

Reply:
I thought we are building nameServiceMap with intersecting hosts. I think we will have to catch the service first with VS hosts and then run this loop. PTAL.